library/src/main/java/software/amazon/location/auth/AuthHelper.kt

```diff
@@ -71,4 +72,25 @@ class AuthHelper(private val context: Context) {
             locationCredentialsProvider // Return the generated locationCredentialsProvider
         }
     }
+
+    /**
+     * Authenticates using a Cognito Identity Pool ID and a specified CredentialsProvider.
+     * @param identityPoolId The identity pool id.
+     * @param credentialsProvider The CredentialsProvider from AWS kotlin SDK.
+     * @return A LocationCredentialsProvider object.
+     */
+    suspend fun authenticateWithCredentialsProvider(
+        identityPoolId: String,
+        credentialsProvider: CredentialsProvider?
+    ): LocationCredentialsProvider {
```

Comment:
The expectation is t authenticate only using credentialsProvider, without identityPoolId

Reply:
Removed